pkg/util/platform_status.go

```diff
@@ -0,0 +1,59 @@
+package util
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetPlatformStatus provides a backwards-compatible way to look up platform status. AWS is the
+// special case. 4.1 clusters on AWS expose the region config only through install-config. New AWS clusters
```

Comment:
@Miciah : do we need to drag the dependency on the cluster config map for the new components?

Reply:
@alebedev87 I think there is no harm to have this.

Reply:
@alebedev87 if I remove it I will again to a lot of testing :) I am using the way we use for cluster ingress operator.  I think we can do such type of refactoring once e2e tests are in place.

Reply:
@alebedev87 I have removed it as per your expectation.

Reply:
@alebedev87 PTAL . Lets merge this if you are happy.

Reply:
The isn't that the operator could be running on an OpenShift 4.1 cluster; the issue is that the operator could be running on a cluster that was installed at version 4.1 and upgraded to 4.10, and such a cluster might not have `PlatformStatus` set.  

That said, OpenShift-specific logic should be in in a separate package as much as possible to avoid having platform-specific logic in the main logic.